cmdstanpy/utils.py

```diff
@@ -128,46 +128,127 @@ def check_csv(filename: str) -> Dict:
 def scan_stan_csv(filename: str) -> Dict:
     """Process stan_csv file line by line."""
     dict = {}
-    draws_found = 0
     lineno = 0
-    with open(filename) as fp:
-        line = fp.readline().strip()
+    try:
+        fp = open(filename, 'r')
+        try:
+            lineno = scan_config(fp, dict, lineno)
+            lineno = scan_column_names(fp, dict, lineno)
+            lineno = scan_metric(fp, dict, lineno)
+            lineno = scan_draws(fp, dict, lineno)
+        finally:
+            fp.close()
+    except IOError:
+        raise IOError()
```

Comment:
Here the same, `with open` is better than try-except

And if not exception handling is done, not try block is needed

Reply:
using `open` per comment.